rca/api_content/content.py

```diff
@@ -134,25 +164,15 @@ def pull_news_and_events(programme_type_slug=None):
     query = query.urlencode()
     events_url = f"{settings.API_CONTENT_BASE_URL}/api/v2/pages/?{query}"
 
-    # By default, work with 3 news items but if we pull events, get 2
-    # so we end up with 2 x News/Blog and 1 x Event.
-    news_items_to_get = 3
-
     events_data = []
-    try:
-        response = requests.get(url=events_url, timeout=5)
-        response.raise_for_status()
-        logger.info("pulling Events from API")
-    except requests.exceptions.HTTPError:
-        error_text = "Error occured when fetching event data"
-        logger.exception(error_text)
-        raise CantPullFromRcaApi(error_text)
-    else:
-        data = response.json()
-
-        if data["meta"]["total_count"] > 0:
-            news_items_to_get = 2
-            events_data = parse_items_to_list(data, "Event")
+    fetched_event_data = fetch_data(events_url)
+    if (
+        fetched_event_data
+        and "total_count" in fetched_event_data["meta"]
+        and fetched_event_data["meta"]["total_count"] > 0
+    ):
+        news_items_to_get = NEWS_ITEMS_WITH_EVENTS
```

Comment:
You shouldn't need to assign the constant to a variable (its just a normal variable anyway)